docs/sources/flow/reference/components/prometheus.exporter.blackbox.md

```diff
@@ -0,0 +1,107 @@
+---
+# NOTE(rfratto): the title below has zero-width spaces injected into it to
+# prevent it from overflowing the sidebar on the rendered site. Be careful when
+# modifying this section to retain the spaces.
+#
+# Ideally, in the future, we can fix the overflow issue with css rather than
+# injecting special characters.
+
+title: prometheus.exporter.blackbox
+---
+
+# prometheus.exporter.blackbox
+The `prometheus.exporter.blackbox` component embeds
+[`blackbox_exporter`](https://github.com/prometheus/blackbox_exporter). This allows
+the collection of blackbox metrics (probes) and exposing them as Prometheus metrics.
+
+## Usage
+
+```river
+prometheus.exporter.blackbox "LABEL" {
+}
+```
+
+## Arguments
+The following arguments can be used to configure the exporter's behavior.
+Omitted fields take their default values.
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`config_file`                 | `string`       | Blackbox configuration file with custom modules | | yes
+`probe_timeout_offset`        | `float`        | Represents the offset to subtract from timeout in seconds when probing targets  | 0.5 | no
+
+If `config_file` is a YAML file with definition of blackbox modules to use. See https://github.com/prometheus/blackbox_exporter/blob/master/example.yml for more details how to generate custom file.
+
+## Blocks
+
+The following blocks are supported inside the definition of
+`prometheus.exporter.blackbox` to configure collector-specific options:
+
+Hierarchy | Name | Description | Required
+--------- | ---- | ----------- | --------
+blackbox_target | [blackbox_target][] | Configures a blackbox target. | yes
+
+[blackbox_target]: #blackbox_target-block
+
+### blackbox_target block
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`name` | `string` | Name of a blackbox_target. | | yes
```

Comment:
Do these names have to be unique? If so, it might make sense to make the name a label so you have 

```river 
blackbox_target "example" {
  address = "..."
}

blackbox_target "grafana" {
  address = ... 
}
```